native/src/routes/intro/IntroContainer.js

```diff
@@ -238,10 +242,10 @@ class Intro extends React.Component<PropsType, StateType> {
 }
 
 const mapStateToProps = (state: ReduxStateType): {| language: string |} => ({ language: state.contentLanguage })
-type ConnectType = {| language: string, dispatch: () => void |}
+type ConnectType = {| ...OwnPropsType, language: string, dispatch: () => void |}
 
-export default connect<ConnectType, {||}, _, _, _, _>(mapStateToProps)(
-  withTranslation(['intro', 'settings'])(
-    withTheme(Intro)
+export default connect<ConnectType, OwnPropsType, _, _, _, _>(mapStateToProps)(
+  withTranslation<$Diff<PropsType, {| theme: ThemeType |}>>(['intro', 'settings'])(
+    withTheme<PropsType>(Intro)
```

Comment:
Actually it's not that complicated. You just have to specify what props the component will receive.Therefore in this case the component wrapped in `withTranslation` will receive all props with exception of the `theme` because that is only injected by `withTheme` later. On the other hand `Intro` wrapped in `withTheme` will receive all props.

I think if we gradually remove unnecessary `withThemes` and switch to hooks, these will vanish sooner or later.